### PR TITLE
Use `.all` in generated policy scopes

### PR DESCRIPTION
**Pull request by iangreenleaf**
The current generated policy scopes return the `scope` object directly. This can cause some weird inconsistent behavior, since this means that `policy_scope(MyKlass.scope_or_relation)` will return a relation-like object, while `policy_scope(MyKlass)` will return a class, which is not a relation and does not respond appropriately to things like `my_policy_scope.to_a` or `MyKlass.where(xyz).merge(my_policy_scope)`.

I think, and the examples in the README agree, that for at least everyone using ActiveRecord the best default is to return `scope.all`. This patch updates the generated policies to use that default. I also updated a spec helper that was using the same pattern—not crucial, but it avoids misleading people and I think I improved the specs at the same time.

### CODE REVIEW

**Comment by timcheadle** [comment]
Just ran into this and came up with the same solution. 👍

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -83,10 +83,21 @@ def policy_class
...
+  def ==(other)
+    original_object == other.original_object
+  end
+end
+
 class CommentPolicy < Struct.new(:user, :comment)
   class Scope < Struct.new(:user, :scope)
     def resolve
-      scope
+      CommentScope.new(scope)
```
Comment (Linuus) [inline spec/spec_helper.rb]: Why did you change this?

Comment (iangreenleaf) [reply spec/spec_helper.rb]: This spec helper was using the old pattern of returning `scope` directly. I didn't think it made sense to leave that as example code when we're discouraging that pattern's use. I updated it to return a simple mock scope object, so we can test that the policy is correctly wrapping the object it's given.

Comment (Linuus) [reply spec/spec_helper.rb]: Hmm, I think returning this CommentScope object makes the test more confusing that to just return the `scope`. In the test we just want to know that the `resolve` method returns the expected value, whatever it is.

Can you revert the `CommentScope` code for now and I'll merge the generator fix?

Comment (Linuus) [reply spec/spec_helper.rb]: Or nah... Let's keep it as it is :) Thanks!

**Comment by Linuus** [comment]
This looks fine to me!

Can you just rebase on master so we get a green Travis build?

**Comment by iangreenleaf** [comment]
Done, thanks for the review!

**Comment by Linuus** [comment]
Thanks!
